operator/pkg/remote/kyma_synchronization_context.go

```diff
@@ -342,7 +354,22 @@ func (c *KymaSynchronizationContext) CreateOrUpdateModuleTemplateCatalog(
 	if create {
 		return c.runtimeClient.Create(ctx, catalog)
 	}
-	return c.runtimeClient.Update(ctx, catalog)
+	if templatesNeedUpdate {
+		return c.runtimeClient.Update(ctx, catalog)
+	}
+	return nil
+}
+
+func DoesModuleTemplateNeedUpdateInCatalog(catalog *corev1.ConfigMap, template *v1alpha1.ModuleTemplate) bool {
+	moduleName := template.GetLabels()[v1alpha1.ModuleName]
+	if catalog.Data[moduleName] != "" {
+		oldCatalogEntry := &CatalogEntry{}
+		if err := yaml.Unmarshal([]byte(catalog.Data[moduleName]), oldCatalogEntry); err != nil ||
```

Comment:
```suggestion
		if err := yaml.Unmarshal([]byte(catalog.Data[moduleName]), oldCatalogEntry); err != nil &&
```

Reply:
I did the Logic OR on purpose as I do not want the Update to be ignored in case one entry is failing. Do you disagree with this?

Reply:
oh, sorry, my mistake, I'm previously thinking this `err != nil` will lead to `oldCatalogEntry.Generation` null pointer, but actually in this case, exactly should use OR.